File: json/json-annotation-processor/src/main/java/ru/tinkoff/kora/json/annotation/processor/extension/JsonLinkerExtensionFactory.java
package ru.tinkoff.kora.json.annotation.processor.extension;

import ru.tinkoff.kora.json.annotation.processor.JsonTypes;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.ExtensionFactory;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.KoraExtension;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public class JsonLinkerExtensionFactory implements ExtensionFactory {
    @Override
    public Optional<KoraExtension> create(ProcessingEnvironment processingEnvironment) {
        var json = processingEnvironment.getElementUtils().getTypeElement(JsonTypes.json.canonicalName());
        if (json == null) {
            return Optional.empty();
        } else {
            return Optional.of(new JsonKoraExtension(processingEnvironment));
        }
    }
}

File: json/json-symbol-processor/src/main/kotlin/ru/tinkoff/kora/json/ksp/extension/JsonLinkerExtensionFactory.kt
package ru.tinkoff.kora.json.ksp.extension

import com.google.devtools.ksp.getClassDeclarationByName
import com.google.devtools.ksp.processing.CodeGenerator
import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.processing.Resolver
import ru.tinkoff.kora.json.ksp.JsonTypes
import ru.tinkoff.kora.kora.app.ksp.extension.ExtensionFactory
import ru.tinkoff.kora.kora.app.ksp.extension.KoraExtension

class JsonLinkerExtensionFactory : ExtensionFactory {
    override fun create(resolver: Resolver, kspLogger: KSPLogger, codeGenerator: CodeGenerator): KoraExtension? {
        val json = resolver.getClassDeclarationByName(JsonTypes.json.canonicalName)
        return if (json == null) {
            null
        } else {
            JsonKoraExtension(resolver, kspLogger, codeGenerator)
        }
    }
}
